core/classes/BugFilterQuery.class.php

```diff
@@ -1406,48 +1439,79 @@ protected function build_prop_search() {
 		}
 
 		# break up search terms by spacing or quoting
-		preg_match_all( "/-?([^'\"\s]+|\"[^\"]+\"|'[^']+')/", $this->filter[FILTER_PROPERTY_SEARCH], $t_matches, PREG_SET_ORDER );
-
-		# organize terms without quoting, paying attention to negation
-		$t_search_terms = array();
-		foreach( $t_matches as $t_match ) {
-			$t_search_terms[trim( $t_match[1], "\'\"" )] = ( $t_match[0][0] == '-' );
-		}
+		# optional "s:, i:, c:" for faster summary only requests
+		preg_match_all( "/-?([sic]:)?([^'\"\s]+|\"[^\"]+\"|'[^']+')/", $this->filter[FILTER_PROPERTY_SEARCH], $t_matches, PREG_SET_ORDER );
 
 		$t_bugnote_table = $this->helper_table_alias_for_bugnote();
 
 		# build a big where-clause and param list for all search terms, including negations
 		$t_first = true;
-		$t_textsearch_where_clause = '( ';
-		foreach( $t_search_terms as $t_search_term => $t_negate ) {
-			if( !$t_first ) {
-				$t_textsearch_where_clause .= ' AND ';
+		$t_after_or = false; # if we are after an OR 
+		$t_textsearch_where_clause = "( ( "; # second ( for OR
+		$t_allow_custom_field_search = ( ON == config_get( 'filter_by_custom_fields' ) );
+
+		foreach( $t_matches as $t_match ) {
+			# analyse search term
+			$t_search_term = trim( $t_match[2], "\'\"" );
+			$t_negate = ( $t_match[0][0] == '-' );
+			$t_summary_only = ( $t_match[1] && $t_match[1][0] == 's' );
+			$t_id_only = ( $t_match[1] && $t_match[1][0] == 'i' );
+			$t_custom_field_only = ( $t_match[1] && $t_match[1][0] == 'c' );
+			if ( !$t_first && !$t_after_or ) {
+				if ( $t_search_term == 'OR' ) {
+					$t_textsearch_where_clause .= ' ) OR ( ';
+					$t_after_or = true;
+					continue;
+				} else {
+					$t_textsearch_where_clause .= ' AND ';
+				}
 			}
 
 			if( $t_negate ) {
 				$t_textsearch_where_clause .= 'NOT ';
 			}
 
 			$c_search = '%' . $t_search_term . '%';
-			$t_textsearch_where_clause .= '( ' . $this->sql_like( '{bug}.summary', $c_search )
-					. ' OR ' . $this->sql_like( '{bug_text}.description', $c_search )
-					. ' OR ' . $this->sql_like( '{bug_text}.steps_to_reproduce', $c_search )
-					. ' OR ' . $this->sql_like( '{bug_text}.additional_information', $c_search )
-					. ' OR ' . $this->sql_like( '{bugnote_text}.note', $c_search );
+			$t_textsearch_where_clause .= '( ';
+			# search in summary only if not id / custom_field_search
+			if ( !$t_id_only && !$t_custom_field_only ) {
+				$t_textsearch_where_clause .= $this->sql_like( '{bug}.summary', $c_search );
+                        } else {
+				$t_textsearch_where_clause .= '0';
```

Comment:
this is not cross-compatible syntax, only works in mysql, but no in other databases: oracle, sql server, postgres

Reply:
What is the cross-compatible syntax?

Reply:
> What is the cross-compatible syntax?

What is exactly the intention?
Ignore one of the clauses in an ORed expression? 
Seems lke the resulkt would be "0 OR this OR that"
Then i would not add the caluse, nor the first OR.

A dirtier hack is to use a more portable boolean expression like "1=1" (true) or "1=0" (false)
but i'd rather manage properly the building of the conditions.

Reply:
This would require to rewrite all "t_textsearch_where_clause .=" lines in this function and in the function above, it would require a different handling of the situation where no custom field is visible. Perhaps this is something a person with better php skills can do? It would probably go along the line of adding to a list and then joining the list with " OR ".

I first would like to know if the change is acceptible anyway. For now I will add the 1=0 hack.

Reply:
> adding to a list and then joining the list with " OR "

Yes, that would be the ideal approach. Build an array and an the end, if not emply, implode with " OR ".
Even, if the item count is 1, omit the outer ( ).